pandas/tests/io/parser/usecols.py

```diff
@@ -481,7 +481,7 @@ def test_raise_on_usecols_names_mismatch(self):
         data = 'a,b,c,d\n1,2,3,4\n5,6,7,8'
 
         if self.engine == 'c':
-            msg = 'Usecols do not match names'
+            msg = "do not match columns, columns expected but not found"
```

Comment:
I think it would be good to have a stronger check for which columns are listed (both in the `self.orig_names` and `self.names` case.

Reply:
@gfyoung I've just tested with the Python engine, and the error is:

```
ValueError: ' column3' is not in list
```

Would we want the error to be consistent between both engines? If yes, would we prefer the proposed or existing format?

Reply:
Your error message is better.  Let's use it for both engines.